### PR TITLE
add attachment and dropbox services and controller

### PR DESCRIPTION
**Pull request by Bohadan-Markatov**

### CODE REVIEW

File: src/main/java/app/controller/AttachmentController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                                                  @RequestParam("taskId") Long taskId) {
+        return attachmentService.create(filePath, taskId);
+    }
+
+    @GetMapping("/{id}")
+    public AttachmentResponseDto getAttachmentById(@PathVariable Long id) {
+        return attachmentService.getById(id);
+    }
+
+    @GetMapping("/{id}/all")
```
Comment (bodya4243) [inline src/main/java/app/controller/AttachmentController.java:35]: maybe add response status here

File: src/main/java/app/controller/AttachmentController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    private final AttachmentService attachmentService;
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping(consumes = "multipart/form-data")
+    public AttachmentResponseDto createAttachment(@RequestParam("filePath") String filePath,
+                                                  @RequestParam("taskId") Long taskId) {
+        return attachmentService.create(filePath, taskId);
+    }
+
+    @GetMapping("/{id}")
```
Comment (bodya4243) [inline src/main/java/app/controller/AttachmentController.java:30]: same here

**Review by bodya4243 — APPROVED** [review]
left 2 minor comments

File: src/main/java/app/service/impl/AttachmentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    private String getFileName(String filePath) {
+        Path path = Paths.get(filePath);
+        return path.getFileName().toString();
+    }
+
+    private void createFolderIfNotExists(Long taskId) {
+        if (!taskRepository.existsById(taskId)) {
+            throw new EntityNotFoundException("Can't find task by id: " + taskId);
+        }
+        if (!attachmentRepository.existsByTaskId(taskId)) {
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/service/impl/AttachmentServiceImpl.java:83]: same

File: src/main/java/app/service/impl/AttachmentServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    @Override
+    public AttachmentResponseDto getById(Long attachmentId) {
+        Attachment attachment = attachmentRepository.findById(attachmentId).orElseThrow(()
+                -> new EntityNotFoundException("Can't find attachment by id: " + attachmentId));
+        return mapToDtoAndSetFileLink(attachment);
+    }
+
+    @Override
+    public List<AttachmentResponseDto> getAllByTaskId(Long taskId) {
+        if (!attachmentRepository.existsByTaskId(taskId)) {
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/service/impl/AttachmentServiceImpl.java:49]: and here

File: src/main/java/app/service/impl/AttachmentServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        return dropboxService.getFilePublicLink(dropBoxFileId);
+    }
+
+    private String getFileName(String filePath) {
+        Path path = Paths.get(filePath);
+        return path.getFileName().toString();
+    }
+
+    private void createFolderIfNotExists(Long taskId) {
+        if (!taskRepository.existsById(taskId)) {
```
Comment (VlasenkoYevhenii) [inline src/main/java/app/service/impl/AttachmentServiceImpl.java:80]: I would rather use positive conditions to make it more readable kinda 
```suggestion
        if (taskRepository.findById().isEmpty()) {

```
or something like that

**Review by VlasenkoYevhenii — APPROVED** [review]

**Pull request merged**